client/concordclient/include/client/concordclient/concord_client_exceptions.hpp

```diff
@@ -40,4 +40,17 @@ class InternalError : public std::runtime_error {
  public:
   InternalError() : std::runtime_error("an internal error occurred"){};
 };
+
+// The stream ended
+class EndOfStream : public std::runtime_error {
```

Comment:
I am thinking that using exceptions for control flow is not ideal - for example, here `EndOfStream` is a successful condition, yet it inherits from `runtime_error`. Is there a way for us to avoid using exceptions for signalling these conditions?

Reply:
This is defined by the container created for ITC. The data is returned using the RemoteRecipient and the state is sent through Exception. So in the current scenario, this is as per implementation. I thought of changing that using a different way by adding a new status enum type variant in RemoteRecipient, but that will require major changes. We can take this in a separate PR.

Reply:
Cool, let's address separately. Thanks!